### DISCUSSION TITLE
TPU on Colab: unexpected keyword argument 'num_tpu_cores'

### DISCUSSION THREAD

**Discussion by naraugialusru** [Lightning-AI/pytorch-lightning#6244] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
On a Colab TPU, I get the error:

```
File "/usr/local/lib/python3.7/dist-packages/pytorch_lightning/trainer/connectors/env_vars_connector.py", line 42, in overwrite_by_env_vars
    return fn(self, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'num_tpu_cores'
```

I'm running a script that is in my Google Drive. I copy the data up onto Colab. When I do the exact same process for the GPU, it works (with Colab set to GPU, and the appropriate trainer parameter adjustment). 

The trainer instantiation is simply

```
    trainer = pl.Trainer(logger=logger,
                         num_tpu_cores=8,
                         fast_dev_run=False,
                         max_epochs=20)
```

**Comment by akihironitta** [comment, marked as answer]
@naraugialusru `num_tpu_cores` should be `tpu_cores`. `num_tpu_cores` was removed in #2760. See also: [the docs](https://pytorch-lightning.readthedocs.io/en/1.2.1/common/trainer.html#tpu-cores)
